docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,121 @@
+- Needs to expose an option for users to opt out of this feature
+
+Additional changes needed:
+- Documentation regarding terms and conditions?
+- Documentation about how to opt out
+- Documentation of requirements: non-disconnected environment
+
+Sending data should not interfere with customer workloads and deployment
+
+### User Stories
+
+#### Story 1
+
+As an infrastructure operator, I can provision and manage my fleet of clusters knowing that data points are automatically collected and sent to the Assisted Installer team without having to do additional configurations. I know my data will be safe and secure and the team will only collect data they need to improve the product. 
+
+
+#### Story 2
+
+As a developer on the assisted installer team, I can analyze the customer data to determine if a feature is worth implementing/keeping/improving. I know that the customer data is accurate and up-to-date. All of the data is parse-able and can be easily tailored to the graphs/visualizations that help my analysis.
+
+#### Story 3
+
+As a product owner, I can determine if the product is moving in the right direction based on the actual customer data. I can help determine which features and bug fixes to prioritize based on the data.
+
+### Implementation Details/Notes/Constraints
+
+Assisted Service changes:
+1. New environment variables to be added:
+    - `DISABLE_DATA_COLLECTION`: If set to `true`, it will disable this feature and thereby allow a customer to opt-out, otherwise the feature is on by default.
+    - `DATA_COLLECTION_URL`: The URL the data will be sent to.
+2. Add additional logic when a cluster reaches one of the following end stages: `error`, `cancelled`, `installed`
+    - Each of these cluster stages represent an end state and that the cluster attempted installation.
+    - Additional logic includes: querying the DB for events of that cluster and streaming this event data to the `DATA_COLLECTION_URL` over HTTPS.
+        - If the customer has large-scale clusters, would it be better to always compress the data in a separate thread?
+3. Add logic for authenticating to the `DATA_COLLECTION_URL` 
+    - Currently unknown since the service to host this data hasn't been decided on yet, but it could take the form of using a Kubernetes Secret
+
+### Risks and Mitigations
+
+On-premises users might have large-scale clusters so network usage and processor usage could be impacted depending on how large the data set is. 
+
+## Design Details
+
+### Open Questions
+
+1. Are we planning to support disconnected environments in the future?
+2. Does any of the event data we store contain Personally Identifiable Information? Does this need to be scrubbed or modified before we push this feature out?
+
+### UI Impact
+
+No UI changes will need to be made.
+
+### Test Plan
+
+Add a check at the end of a cluster install attempt that will query the correct data made it to the data hosting service if the feature is enabled.
```

Comment:
check it with large-scale cluster - more than 100 hosts

Reply:
Good point! Added to the test plan - how does it sound?